yt_dlp/utils/traversal.py

```diff
@@ -168,14 +169,43 @@ def apply_key(key, obj, is_last):
                 result = next((v for k, v in obj.groupdict().items() if casefold(k) == key), None)
 
         elif isinstance(key, (int, slice)):
-            if is_iterable_like(obj, collections.abc.Sequence):
+            if is_iterable_like(obj, collections.abc.Sequence) or isinstance(obj, xml.etree.ElementTree.Element):
                 branching = isinstance(key, slice)
                 with contextlib.suppress(IndexError):
                     result = obj[key]
             elif traverse_string:
                 with contextlib.suppress(IndexError):
                     result = str(obj)[key]
 
+        elif isinstance(obj, xml.etree.ElementTree.Element) and isinstance(key, str):
+            xpath, sep, special = key.rpartition('/')
+            has_specials = special.startswith('@') or special == 'text()'
+            if not has_specials:
+                xpath = f'{xpath}{sep}{special}'
```

Comment:
How about:

```suggestion
            if not special.startswith('@') or special != 'text()'
                xpath, special = key, None
```

This way, there's no need for separate `special` and `has_special`

Reply:
the regex to do this is way less readable than using partition:
```py
xpath, special = re.fullmatch(r'(.*)(?:/(@\w*|text\(\)))?', key).group(1, 2)
```
```py
xpath, _, special = key.partition('/')
if not special.startswith('@') or special != 'text()':
    xpath = key
    special = None
```

Reply:
I was considering doing the `./` normalization also in same regex. But I agree your code's more readable. Hence why it was just a side note